fix: invalid border-width check

fu050409 opened the pull request:
Previous: `const hasBorder = borderStyle && borderStyle !== 'none';`

Now: `const hasBorder = borderStyle && borderStyle !== 'none' && parseFloat(style.getPropertyValue('border-width')) > 0;`

In the current code, even if the border style is set to `hidden` or `transparent`, it will still be matched. I added the `border-width` check to solve this problem.

tinchox5 reviewed (commented):
@fu050409 thanks for the suggestion! Just one think you may test before proceed... could you use the cache? I dont want to use getPropertyValue if is possible to use the cache to avoid a new computation.

tinchox5 commented:
@fu050409 what do you think? 

```
const borderWidth = parseFloat(style.borderWidth);

const hasBorder =
  borderStyle &&
  borderStyle !== "none" &&
  borderStyle !== "hidden" &&
  borderWidth > 0;
```

I mean... style.borderWidth accesses the computed property directly as an attribute of the CSSStyleDeclaration object, avoiding a string lookup.

fu050409 commented:
@tinchox5 I looked up some documents, according to the description of [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Window/getComputedStyle#return_value), Using `getPropertyValue` should have the same effect as directly accessing this property. The only difference might be that we can use the more elegant hump nomenclature?

tinchox5 commented:
Ok thanks for sharing this. Anyway, in this case I'd prefer to use style.borderWidth. If you could change it I will merge it!

fu050409 commented:
> Ok thanks for sharing this. Anyway, in this case I'd prefer to use style.borderWidth. If you could change it I will merge it!

Sure! The reason why I chose this approach is that `getPropertyValue` was used in the previous code. To maintain a consistent style, I also used `getPropertyValue`. Personally, I also prefer the form of directly accessing attributes and using hump naming, which is more elegant.

Do you want me to change the existing code to hump naming style as well?

tinchox5 commented:
Hump naming style not necessary this stage

fu050409 commented:
@tinchox5 Hi, sorry for my late response since my city was hit by a typhoon yesterday, lol.

> Hump naming style not necessary this stage

Sorry, I didn't fully understand what do you mean. Could you please clarify?

Now I have already removed all the `getPropertyValue` and replaced them with query properties directly. For example, `style.getPropertyValue('border-width')` to `style.borderWidth`. Could you please review the code to see if it meets your expectations?

fu050409 commented:
> OMG are you ok? Sorry to hear that

Thank you for your concern. We just had a very strong wind and rain, and there was no actual loss.

I found a few more issues while browsing through the snapDOM source code, and the issue with image scaling seems to be popping up again on iOS (currently on iOS 18), so I might have a few more patch PRs later.
